Src/Coravel/Scheduling/Schedule/Scheduler.cs

```diff
@@ -41,6 +42,30 @@ public IScheduleInterval ScheduleAsync(Func<Task> asyncTaskToSchedule)
             return scheduled;
         }
 
+        public IScheduleInterval Schedule(Action actionToSchedule, Func<bool> customRestriction)
```

Comment:
This would break the fluent nature of the exposed API.  Exposing a method on `IScheduledRestriction` would make more sense since this interface is specifically for exposing any restriction abilities of the scheduled task.

The scheduler itself shouldn't be responsible for handling restrictions.

I think exposing a `When(Func<bool> predicate)` method from IScheduledRestriction would fit closer to the usage of the Scheduler and to the exact purpose of `IScheduledRestriction`.

Does that make sense?

Reply:
Yes it does. Will seek to it.